### ISSUE TITLE
Cannot remove contributor "Testing"

### ISSUE THREAD

**Issue by seabelis** [internetarchive/openlibrary#1958]
## Description
> What problem are we solving? What does the experience look like today? What are the symptoms?
Cannot remove "testing" from the list of contributors.

## Evidence 
Removing it from the edit page behaves normally, but contributor remains after save.
> Relevant url?
https://openlibrary.org/books/OL22288398M/To_Kill_a_Mockingbird

> Screenshot (if possible):

## Expectation
> What should by happening? What will it look like / how will it behave?
Normally contributors can be deleted.

## Details

> Logged in (Y/N)?

> Browser type/version?
Chrome Version 72.0.3626.119 (Official Build) (64-bit)

> Operating system?
Win 7 Pro
## Proposal & Constraints

> What is the proposed solution / implementation? Is there a precedent of this approach succeeding elsewhere?

> Which suggestions or requirements should be considered for how feature needs to appear or be implemented?

## Stakeholders
> @ tag stakeholders of this bug / feature

**Comment by tfmorris** [comment]
I can confirm/reproduce this. The form data looks a little weird, but I can't tell if it's actually wrong without more investigation:

> role: 
> name: 
> {{prefix}}contributors--{{index}}--role: {{role}}
> {{prefix}}contributors--{{index}}--name: {{name}}

The edit request is getting sent, but is neither creating a new version nor generating an error (it returns a 303 See Other).

**Comment by brad2014** [comment]
@seabelis, can you do me a favor and check if the bug is "Deleting contributor is broken" (you perform the deletion but upon refresh the contributor is still there), or is it "Deleting contributor works unless the contributor is Testing"?

@cdrini, could this be a caching problem?  (we've seen a few bugs like something is deleted, but still hanging out in the cache)

**Comment by cdrini** [comment]
I don't think this is a caching issue; that's usually the case when deleting Things (i.e. works/authors/books/etc). This is about deleting a field on a thing.

**Comment by seabelis** [comment]
@brad2014 The name "Testing" seems to be the problem. I've tested it by adding it to this record and now I cannot remove it. https://openlibrary.org/books/OL25003949M/The_complete_idiot's_guide_to_world_religions

**Comment by guyjeangilles** [comment]
Still a bug for the linked pages.

**Comment by seabelis** [comment]
Now that I can revert to previous versions I can fix these. Not ideal since reverting reverts _all_ changes from that point meaning any good edits need to be reapplied manually.

**Comment by tfmorris** [comment]
I added test2 to the new example https://openlibrary.org/books/OL25003949M/The_complete_idiot's_guide_to_world_religions and then was able to remove "Testing" so I think that the bug is actually that you can't remove the last contributor, not anything to do with the name of the contributor.

**Comment by seabelis** [comment]
Ahh. Yes, I see. You're right.

**Comment by tfmorris** [comment]
Any edit that results in an empty contributors list (ie removing the only two contributors) results in no change. It looks like the backend must be interpreting the empty payload as "no change" rather than "empty list"
